backend/src/sms/routes/tests/sms-callback.routes.test.ts

```diff
@@ -0,0 +1,168 @@
+let credential: Credential
+
+const app = initialiseServer(false)
+
+beforeEach(async () => {
+  user = await User.create({
+    email: 'sms_callback@agency.gov.sg',
+  } as User)
+  const userId = user.id
+  apiKey = await user.regenerateAndSaveApiKey()
+  credential = await Credential.create({ name: 'twilio' } as Credential)
+  await UserCredential.create({
+    label: `twilio-callback-${userId}`,
+    type: ChannelType.SMS,
+    credName: credential.name,
+    userId,
+  } as UserCredential)
+})
+
+beforeAll(async () => {
+  sequelize = await sequelizeLoader(process.env.JEST_WORKER_ID || '1')
+})
+
+afterEach(async () => {
+  jest.clearAllMocks()
+  await SmsMessageTransactional.destroy({ where: {} })
+  await User.destroy({ where: {} })
+  await UserCredential.destroy({ where: {} })
+  await Credential.destroy({ where: {} })
+})
+
+afterAll(async () => {
+  await sequelize.close()
+  await (app as any).cleanup()
+})
+
+describe('On successful message send, status should update according to Twilio response', () => {
+  const validApiCall = {
+    body: 'Hello world',
+    recipient: '98765432',
+    label: 'twilio-callback-1',
+  }
+  test('Should send a message successfully', async () => {
+    const mockSendMessageResolvedValue = 'message_id_callback'
+    const mockSendMessage = jest
+      .spyOn(SmsService, 'sendMessage')
+      .mockResolvedValue(mockSendMessageResolvedValue)
+    mockSecretsManager.getSecretValue().promise.mockResolvedValueOnce({
+      SecretString: JSON.stringify(TEST_TWILIO_CREDENTIALS),
+    })
+    const res = await request(app)
+      .post('/transactional/sms/send')
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send(validApiCall)
+
+    expect(res.status).toBe(201)
+    expect(mockSendMessage).toBeCalledTimes(1)
+
+    const transactionalSms = await SmsMessageTransactional.findOne({
+      where: { userId: user.id.toString() },
+      order: [['createdAt', 'DESC']],
+    })
+    const transactionalSmsId = transactionalSms?.id
+
+    const getByIdRes = await request(app)
+      .get(`/transactional/sms/${transactionalSmsId}`)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send()
+    expect(getByIdRes.status).toBe(200)
+    expect(getByIdRes.body.status).toBe(TransactionalSmsMessageStatus.Unsent)
+    expect(getByIdRes.body.body).toEqual('Hello world')
+    expect(getByIdRes.body.recipient).toEqual('98765432')
+    expect(getByIdRes.body.credentialsLabel).toEqual('twilio-callback-1')
+
+    expect(getByIdRes.status).toBe(200)
+    expect(getByIdRes.body.status).toBe(TransactionalSmsMessageStatus.Unsent)
+    const sampleTwilioCallback = {
+      SmsSid: mockSendMessageResolvedValue,
+      SmsStatus: 'sent',
+      MessageStatus: 'sent',
+      To: '+1512zzzyyyy',
+      MessageSid: mockSendMessageResolvedValue,
+      AccountSid: 'ACxxxxxxx',
+      From: '+1512xxxyyyy',
+      ApiVersion: '2010-04-01',
+      ErrorCode: 'ERRORBOI',
+    }
+    sampleTwilioCallback.ErrorCode = ''
```

Comment:
**[nit / change request]**

actually just wondering, why not instead of mutating the object, we create new object with different error codes? that way it's clearer.
so what I'd do is:
- default `sampleTwilioCallback` is the happy path one
- new `sampleTwilioCallbackError` contains error code, e.g.

```JS
const sampleTwilioCallbackError = {
      ...sampleTwilioCallback
      MessageStatus: 'failed',
      ErrorCode: 'ERRORBOI',
}
```

think it's easier to follow that mutating the object!

Reply:
good idea, completely forgot about spread, changed